app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/AdditionalOptions.kt

```diff
@@ -172,3 +200,51 @@ fun AttachmentAndAdditionalOptionsMenuItems(
         )
     }
 }
+
+@Composable
+fun FragmentContainer(
+    modifier: Modifier = Modifier,
+    fragmentManager: FragmentManager,
+    commit: FragmentTransaction.(containerId: Int) -> Unit
+) {
+    val containerId by rememberSaveable { mutableStateOf(View.generateViewId()) }
+
+    AndroidView(
+        modifier = modifier,
+        factory = { context ->
+            fragmentManager.findFragmentById(containerId)?.view
+                ?.also { (it.parent as? ViewGroup)?.removeView(it) }
+                ?: FragmentContainerView(context)
+                    .apply { id = containerId }
+                    .also {
+                        fragmentManager.commit { commit(it.id) }
+                    }
+        },
+        update = {}
+    )
+}
+
+@Composable
+fun <T : ViewBinding> FragmentHolderScreen(
+    androidViewBindingFactory: (inflater: LayoutInflater, parent: ViewGroup, attachToParent: Boolean) -> T,
+    onBackPress: () -> Unit = {},
+    androidViewBindingUpdate: T.() -> Unit = {},
+) {
+    WireScaffold(topBar = {
+        WireCenterAlignedTopAppBar(
+            elevation = 0.dp,
+            title = "",
+            navigationIconType = NavigationIconType.Close,
+            onNavigationPressed = onBackPress
+        )
+//        } else {
+//            Spacer(modifier = Modifier.height(MaterialTheme.wireDimensions.welcomeVerticalPadding))
+//        }
+    }) { paddingValues ->
+        AndroidViewBinding(
+            factory = androidViewBindingFactory,
+            modifier = Modifier.padding(paddingValues),
+            update = androidViewBindingUpdate,
+        )
+    }
+}
```

Comment:
Please Ignore. This is removed in part 2. 
This was part of the prev spike work.

Reply:
removed